cmd/generate/generate.go

```diff
@@ -123,12 +131,17 @@ func NewCmdCommatrixGenerate(cs *client.ClientSet, streams genericiooptions.IOSt
 			return nil
 		},
 	}
+
 	cmd.Flags().StringVar(&o.destDir, "destDir", "", "Output files dir (default communication-matrix)")
 	cmd.Flags().StringVar(&o.format, "format", "csv", "Desired format (json,yaml,csv,nft)")
-	cmd.Flags().BoolVar(&o.debug, "debug", false, "Debug logs")
+	cmd.Flags().StringVar(&o.platformType, "platform-type", "", "Platform Type (baremetal, aws). Required")
 	cmd.Flags().StringVar(&o.customEntriesPath, "customEntriesPath", "", "Add custom entries from a file to the matrix")
 	cmd.Flags().StringVar(&o.customEntriesFormat, "customEntriesFormat", "", "Set the format of the custom entries file (json,yaml,csv)")
 	cmd.Flags().BoolVar(&o.openPorts, "host-open-ports", false, "Generate communication matrix, host open ports matrix, and their difference")
+	cmd.Flags().BoolVar(&o.debug, "debug", false, "Debug logs")
+	if err := cmd.MarkFlagRequired("platform-type"); err != nil { // make it as required flag
+		log.Fatalf("failed to mark platform-type as required: %v", err)
+	}
```

Comment:
I think this can be removed? we have the check beneath

Reply:
i want to separate the error msg if it invalid or missing